tests/tr1/tests/fstream2/test.cpp

```diff
@@ -5,15 +5,14 @@
 #define TEST_NAME "<fstream>, part 2"
 
 #include "tdefs.h"
+#include "temp_file_name.h"
 #include <assert.h>
 #include <fstream>
 #include <wchar.h>
 
-#undef tmpnam
-#define tmpnam(x) _tempnam(".", "")
-
 void test_main() { // test basic workings of wide fstream definitions
-    const char* tn = CSTD tmpnam(0);
+    const auto temp_name = temp_file_name();
+    const char* tn       = temp_name.c_str();

```

Comment:
this might be unnecessary, streams have string constructor

Reply:
I believe this is by design - the tests are attempting to exercise both constructors.